### ISSUE TITLE
Use DarcyLMH in HM coupling

### ISSUE THREAD

**Issue by jstebel** [flow123d/flow123d#1420]
Currently for some historic reason, the hydro-mechanical model is using the Richards flow equation, which is confusing for users as well as for developers. Change the type of equation to DarcyLMH when possible.

**Comment by jbrezmorf** [comment]
Can we let user to choose one or other Darcy solver as in the Sequential coupling?

**Comment by jstebel** [comment]
DarcyMH is not suitable for transient simulations, therefore only DarcyLMH (or RichardsLMH)is relevant, in my opinion.

**Comment by jbrezmorf** [comment]
Correct, DarcyMH will be removed in release 4.0. Use flow equation factory similar to sequential couplng.

**Comment by jstebel** [comment]
Update: Currently the flow equation in HM coupling can only be DarcyLMH (although the input record for RichardsLMH is also accepted by the program).

The aim is to allow both Darcy and Richards. For this purpose, we need:
- a class with abstract record, which is the base for Darcy and Richards
- this class must be different from DarcyFlowInterface, which is base for Darcy, Richards and HM_Iterative (otherwise the program fails during static initialization due to a loop in the input tree)
